exercises/practice/protein-translation/protein_translation_test.go

```diff
@@ -16,152 +16,149 @@ func TestErrorsNotNil(t *testing.T) {
 type codonCase struct {
 	input         string
 	expected      string
-	errorExpected error
+	expectedError error
 }
 
 var codonTestCases = []codonCase{
 	{
-		"AUG",
-		"Methionine",
-		nil,
+		input:         "AUG",
+		expected:      "Methionine",
+		expectedError: nil,
 	},
 	{
-		"UUU",
-		"Phenylalanine",
-		nil,
+		input:         "UUU",
+		expected:      "Phenylalanine",
+		expectedError: nil,
 	},
 	{
-		"UUC",
-		"Phenylalanine",
-		nil,
+		input:         "UUC",
+		expected:      "Phenylalanine",
+		expectedError: nil,
 	},
 	{
-		"UUA",
-		"Leucine",
-		nil,
+		input:         "UUA",
+		expected:      "Leucine",
+		expectedError: nil,
 	},
 	{
-		"UUG",
-		"Leucine",
-		nil,
+		input:         "UUG",
+		expected:      "Leucine",
+		expectedError: nil,
 	},
 	{
-		"UCG",
-		"Serine",
-		nil,
+		input:         "UCG",
+		expected:      "Serine",
+		expectedError: nil,
 	},
 	{
-		"UAU",
-		"Tyrosine",
-		nil,
+		input:         "UAU",
+		expected:      "Tyrosine",
+		expectedError: nil,
 	},
 	{
-		"UAC",
-		"Tyrosine",
-		nil,
+		input:         "UAC",
+		expected:      "Tyrosine",
+		expectedError: nil,
 	},
 	{
-		"UGU",
-		"Cysteine",
-		nil,
+		input:         "UGU",
+		expected:      "Cysteine",
+		expectedError: nil,
 	},
 	{
-		"UGG",
-		"Tryptophan",
-		nil,
+		input:         "UGG",
+		expected:      "Tryptophan",
+		expectedError: nil,
 	},
 	{
-		"UAA",
-		"",
-		ErrStop,
+		input:         "UAA",
+		expected:      "",
+		expectedError: ErrStop,
 	},
 	{
-		"UAG",
-		"",
-		ErrStop,
+		input:         "UAG",
+		expected:      "",
+		expectedError: ErrStop,
 	},
 	{
-		"UGA",
-		"",
-		ErrStop,
+		input:         "UGA",
+		expected:      "",
+		expectedError: ErrStop,
 	},
 	{
-		"ABC",
-		"",
-		ErrInvalidBase,
+		input:         "ABC",
+		expected:      "",
+		expectedError: ErrInvalidBase,
 	},
 }
 
 func TestCodon(t *testing.T) {
-	for _, test := range codonTestCases {
-		actual, err := FromCodon(test.input)
-		if test.errorExpected != nil {
-			if test.errorExpected != err {
-				t.Fatalf("FAIL: Protein translation test: %s\nExpected error: %q\nActual error: %q",
-					test.input, test.errorExpected, err)
+	for _, tc := range codonTestCases {
+		t.Run(tc.input, func(t *testing.T) {
+			got, err := FromCodon(tc.input)
+			switch {
+			case tc.expectedError != nil:
+				if err == nil {
+					t.Fatalf("FromCodon(%q) expected error: %v, got nil", tc.input, tc.expectedError)
+				}
```

Comment:
The new version seems to be be missing the check that if there was an error and we expected an error, that is is actually the right one as we have ErrStop and ErrInvalidBase above (line 100 in the old code).